### PR TITLE
Simplify feature tile server implementations

### PR DESCRIPTION
**Pull request by eugener**
- Introduce generic `HttpTileRetriever` and `CachedHttpTileRetriever`
- Refactor `CachedOsmTileRetriever` to use `CachedHttpTileRetriever.`

### CODE REVIEW

File: maps/src/main/java/com/gluonhq/impl/maps/tile/CachedHttpTileRetriever.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                cacheThread.start();
+            }
+            logger.info("hasfilecache = " + hasFileCache);
+        } catch (IOException ex) {
+            hasFileCache = false;
+            logger.log(Level.SEVERE, null, ex);
+        }
+    }
+
+    private final static Executor EXECUTOR = Executors.newFixedThreadPool(2, runnable -> {
```
Comment (johanvos) [inline maps/src/main/java/com/gluonhq/impl/maps/tile/CachedHttpTileRetriever.java:89]: 2 can be a System.getProperty() that defaults to 2?

Comment (eugener) [reply maps/src/main/java/com/gluonhq/impl/maps/tile/CachedHttpTileRetriever.java:89]: Yes, that was in the code before I refactored, so I left it unchanged.

Comment (eugener) [reply maps/src/main/java/com/gluonhq/impl/maps/tile/CachedHttpTileRetriever.java]: Updated. It is now retrieved using `gluon.maps.thread.number` system property with the default of `2`

**Comment by eugener** [comment]
Closing as it conflicts with the later PRs

**Pull request closed without merging**